### ISSUE TITLE
Case sensitivity for indent

### ISSUE THREAD

**Issue by machsix** [krvajal/vscode-fortran-support#115]
https://github.com/krvajal/vscode-fortran-support/blob/4f17f244280539ab8cc2448970c5c4b10cadd5f4/language-configuration.json#L53-L54

The current indent rule doesn't support uppercase

**Comment by pedro-ricardo** [comment]
Thanks for pointing that out.
Seems like we have to add a case insensitive option on those regex rules.

**Comment by pedro-ricardo** [comment]
Looks like it does't accept the `(?i)` option of Regex and I can't find a way to include de `/i` flag.
The stupid way to solve this is to double the words with their uppercase equivalent...
```
"increaseIndentPattern": "(then(\\s*|\\s*!.*)$|^\\s*(program|subroutine|function|module|do|block|associate|case|select\\s*case)\\b.*$|\\s*(else|else\\s*if|elsewhere)\\b.*$)|(THEN(\\s*|\\s*!.*)$|^\\s*(PROGRAM|SUBROUTINE|FUNCTION|MODULE|DO|BLOCK|ASSOCIATE|CASE|SELECT\\s*CASE)\\b.*$|\\s*(ELSE|ELSE\\s*IF|ELSEWHERE)\\b.*$)",
"decreaseIndentPattern": "(^\\s*end\\s*(select|if|do|function|subroutine|module|program)\\b.*$|^\\s*else|case\\b.*$)|(^\\s*END\\s*(SELECT|IF|DO|FUNCTION|SUBROUTINE|MODULE|PROGRAM)\\b.*$|^\\s*ELSE|CASE\\b.*$)"
```
It works but it's just wrong to do it that way. I'll keep looking.

**Comment by pedro-ricardo** [comment]
Got it. It has to split the enty of `increaseIndentPattern` into 2 `{"pattern": "string", "flags": "string"}`
Fixed in #117.

**Comment by machsix** [comment]
Thanks! 

B.T.W. where did you find this **stupid** option? :cry: I didn't find any official documentation for the flavor of regex in vscode.

**Comment by pedro-ricardo** [comment]
hehehe ... Ofiicial documentation I never heard of =].
The **stupid** option I mentioned was just to replicate the keywords as lower and upper cases, nothing new. But that leads to a terrible maintence and it just look's like bad programing.

The option I programed, I found in some random issue of VSCode I don't recall now, but in there someone from Microsoft commented that this syntax is possible:
```json
"increaseIndentPattern": {
            "pattern": "then(\\s*|\\s*!.*)$|^\\s*(program|subroutine|function|module|do|block|associate|case|select\\s*case)\\b.*$|\\s*(else|else\\s*if|elsewhere)\\b.*$",
            "flags": "i"
        },
```
That way is the **correct one**, now we keep the same keywords and add the case insensitive flag.
But again .... I didn't find any official documentation on the subject ^.^
